### PR TITLE
Fixing compiler warnings & refactor IsPostbackValid method

### PR DESCRIPTION
**Pull request by Nuss9**

### CODE REVIEW

File: src/SignhostAPIClient/Rest/SignhostApiReceiver.cs [hunk 1]
```diff
@@ -50,8 +50,8 @@ public SignhostApiReceiver(SignhostApiReceiverSettings receiverSettings)
 			}
 
 			if (!string.IsNullOrEmpty(postbackTransaction.Id) &&
-				!string.IsNullOrEmpty(postbackChecksum)
-			) {
+				!string.IsNullOrEmpty(postbackChecksum))
+			{
```
Comment (MrJoe) [inline src/SignhostAPIClient/Rest/SignhostApiReceiver.cs]: Sorry this doesn't match our coding-standards / guidelines, we require the openening `{` to be on the same line as the `)`, sometimes for readability we therefore choose to place the `) {` on a newline.

File: src/SignhostAPIClient/Rest/SignHostApiClient.cs [hunk 2]
```diff
@@ -15,8 +15,7 @@ namespace Signhost.APIClient.Rest
 	/// an signhost api client implementation.
 	/// </summary>
 	public class SignHostApiClient
-		: ISignHostApiClient
-		, IDisposable
+		: ISignHostApiClient, IDisposable
```
Comment (MrJoe) [inline src/SignhostAPIClient/Rest/SignHostApiClient.cs]: Doesn't match our code-guidelines, we require every inheritance to be on its own line and it looks like the original is actually our preferred way of doing things.

File: src/SignhostAPIClient/Rest/SignhostApiReceiver.cs [hunk 3]
```diff
@@ -35,36 +35,63 @@ public SignhostApiReceiver(SignhostApiReceiverSettings receiverSettings)
...
+		}
+
+		private bool ChecksumCalculationParametersAreValid(string postbackChecksum, PostbackTransaction postback)
+		{
+			if (!string.IsNullOrWhiteSpace(postbackChecksum) && !string.IsNullOrWhiteSpace(postback.Id)) {
+				return true;
+			}
+			else {
+				return false;
+			}
```
Comment (MrJoe) [inline src/SignhostAPIClient/Rest/SignhostApiReceiver.cs]: This one can retry the result immediatly there is no need for a boolean function to check if something is true to return true instead you can just return the check immediatly eg:
`return string.IsNullOrWhiteSpace(postbackChecksum) && !string.IsNullOrWhiteSpace(postback.Id);`

and if required you can shorten the signature further to:
```
private bool ChecksumCalculationParametersAreValid(string postbackChecksum, PostbackTransaction postback)
		=> !string.IsNullOrWhiteSpace(postbackChecksum) && !string.IsNullOrWhiteSpace(postback.Id);
```


ps. I find the naming of `ChecksumCalculationParametersAreValid` confusing as I am expecting method parameters / arguments to be validated (which I might not want as an external call), however instead it appears to be something like HasValidChecksumProperties (still not the best name though).

Comment (Nuss9) [reply src/SignhostAPIClient/Rest/SignhostApiReceiver.cs]: I'll implement your suggestion to complete the commit. 
While speculating, I came across this 'Guard Clauses', which might be an interesting alternative:
[https://www.nuget.org/packages/Ardalis.GuardClauses](url)

File: src/SignhostAPIClient/Rest/SignhostApiReceiver.cs (hunk 1 above)
Comment (Nuss9) [reply src/SignhostAPIClient/Rest/SignhostApiReceiver.cs]: Stylecop code analyzer rules and code guidelines differ from one another.
Dropped this commit.

File: src/SignhostAPIClient/Rest/SignHostApiClient.cs (hunk 2 above)
Comment (Nuss9) [reply src/SignhostAPIClient/Rest/SignHostApiClient.cs]: Stylecop code analyzer rules and code guidelines differ from one another.
Dropped this commit.

**Comment by Nuss9** [comment]
Upon review implemented suggestion in SignhostApiReceiver class, and also fixed two code smells there.
Seen the differences between the code guidelines and the code analyzer rules, two commits have been dropped. 
Feel free to merge or cherry-pick from here on.

**Pull request merged**